pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -417,6 +417,12 @@ public synchronized void includeServerToRouting(String instanceId) {
    * Builds/rebuilds the routing for the given table.
    */
   public synchronized void buildRouting(String tableNameWithType) {
+    // skip route building for logical tables
+    if (ZKMetadataProvider.isLogicalTableExists(_propertyStore, tableNameWithType)) {
```

Comment:
I presume this will only happen if a user explicitly calls the `PUT /routing/{tableName}` broker API with a logical table name since we won't have actual state transitions for logical tables that could trigger this routing update?

Reply:
@yashmayya This is triggered with table creation and update too. This is invoked for any change in the brokerResource and refresh table config too.

Reply:
@abhishekbafna I think it will help if you paste example IS/EV after this change. 
Details: https://docs.google.com/document/d/1iS0wtG_V2-W9sQsKkziMjTljT5qKAH_hhsksXr1Tvys/edit?tab=t.0#heading=h.vf97hyarsl5h

tl;dr:
A new entry is made in `brokerResource` for logical tables with the list of applicable brokers and whether they are online/offline. This call is made from the `BrokerResourceOnlineOfflineStateModel`. Since logical tables do not have any segments, this call has to be short-circuited.

Reply:
@abhishekbafna logical table create/update should not trigger this call though ?

Reply:
Looks like the table config refresh message handler in the brokers also triggers this call, and the message is sent by controllers through Helix when a table config is updated. Is this flow different for logical tables?

Reply:
@vrajat create (potentially update when broker tenant is updated) would trigger this. As logical table broker tenant info is store into the brokerResource node and update to the node would trigger it.

![Uploading Screenshot 2025-05-08 at 1.45.56 PM.png…]()

Reply:
Yes. This is called from the BrokerResourceOnlineOfflineStateModel.

Reply:
ack! I misunderstood your statement.

Reply:
I don't think this handling is needed. There are 2 triggering point for this method:
1. BrokerResponse Helix resource (logical table shouldn't be added into it)
2. Physical table config update (logical table config update shouldn't trigger it)

The argument contains type suffix, which most likely won't be the actual logical table name. I'd suggest removing it and simply let table config existence check fail if it is really triggered by a logical table